crates/agglayer-interop-types/src/aggchain_proof.rs

```diff
@@ -26,6 +26,25 @@ pub enum AggchainData {
         /// Optional aggchain proof public values.
         public_values: Option<Box<AggchainProofPublicValues>>,
```

Comment:
Could we refactor `AggchainData::Generic` to be `AggchainProof` + `Signature` - or it would complicate much backward compatibility?

Reply:
it would bring some incompatibility with previous versions IIRC